### DISCUSSION TITLE
Is to continue supporting old linux versions?

### DISCUSSION THREAD

**Discussion by oleg68** [GrandOrgue/grandorgue#1334] [General]
Github announced [swittching Ubuntu 18.04 workers off since 2023-04-01](https://github.com/actions/runner-images/issues/6002). The oldest available version will be 20.04.

GrandOrgue is built on github workers. Building it on 20.04 will require a user linux machine to have glibc 3.31+ and users with other glibc version will be unable to run GrandOrgue. [The table](https://repology.org/project/glibc/versions) shows what linux distributions what glibc version have.

So we have several options:

1. To switch GO builds to 20.04 and to stop supporting old linux versions (including Debian 10 and the old raspbian).
2. To build GrandOrgue in an Ubuntu 18.04 container running on a 20.04 runner with podman.  It will make the build process more complex.
3. To build GO somewhere outside github.
4. Other solutions?

Which options are acceptable for users?

@rousseldenis could you ask this question in the facebook group?

**Comment by rousseldenis** [comment]
@oleg68 yes I do

**Comment by thbeckpr** [comment]
Why continue to support old versions of Linux?

**Reply by rousseldenis** [reply]
Because some users may have old linux distributions and refrain to upgrade.

But I agree that we maybe should align on what is commonly admitted and release cycles. For instance Ubuntu 18.04 LTS maintenance is stopped on 2023-04-30.

@oleg68 We maybe want to focus on fixes and features than on maintenance.

**Reply by larspalo** [reply]
@oleg68 I agree with @rousseldenis and think that we should not worry about providing builds for obsolete versions but just follow along with the release cycles and the tools offered to us. If anyone chooses to stay on an old (obsolete) version they may do so on their own will and also take full responsibility for it.

**Reply by thbeckpr** [reply]
I'm not a GO developer but I agree with Lars.

**Reply by jason-barba** [reply]
I second @thbeckpr's comment.

**Reply by klangpost** [reply]
The AppImage should run everywhere, right?
